package.json

```diff
@@ -84,6 +84,7 @@
     "mssql": "^4.3.2",
     "mysql": "^2.15.0",
     "mysql2": "^1.6.5",
+    "oracledb": "^5.0.0",
```

Comment:
The reason why we didn't have this before in `package.json` is because it will cause many problems to users who want to pull typeorm source, simply run `npm i` without any headache and continue tweaking things.

Is it still problematic to install this package? If yes, I recommend to remove it, and manually add `npm i oracledb` command to CI config

Reply:
I did not have any issues installing this - at least, no more than sqlite3 with node-gyp.  I didn't have to install anything extra.

But - that's just me.  I tried on Linux and a Mac running high sierra (with homebrew).  Is there any way we can test that further?

Reply:
I'm talking about these steps:

https://oracle.github.io/node-oracledb/INSTALL.html#-2-quick-start-node-oracledb-installation

From what I remember it was required to do all these steps otherwise `npm i` on typeorm root would fail. That's why I'm talking, if it still necessary to do these steps for users, it's better to remove oracledb from dependencies.

Reply:
The steps required for the install to work only apply if there's no prebuilt binary.

This does not include fetching the client libraries and setting up the ld config path which is only required if you are actually exercising the OracleDB module.

Instead, the steps include installing dependencies for building a nodejs binary extension using gyp - but ONLY if your architecture and node version does not have a prebuilt binary.  The same requirement applies for sqlite3 as it's also a binary module.

If this is a concern we can move this to a peer or optional dependency?  But then the same should be done for sqlite3 & better-sqlite.

Reply:
I didn't have any problems installing sqlite3 & better-sqlite.... Don't remember any at least. But oracle definitely is a problem to install. 

Anyway, my point is - any user should be able to just clone the repo, run `npm i` and just use it. With oracle its complicated (and we had users complaining about it few years ago). And since it looks like nothing has changed, and it's still a problem. I didn't heard anything about sqlite from users regarding to its issues.

So I suggest to remove it, and make oracle tests to run only on CI, so it won't bring problems to users on their local machines.

Reply:
Sure, I can do that.  I'll get that later today.  And as it stands, oracle tests wouldn't run locally unless you enabled them.

Also, the issues you're probably referring to are that the client libraries were needed to import the module before v3.1.0.  However, that's been changed to not be the case

The v4 release brought a refactor to use a different node API so the builds are portable allowing for a binary distributable to be used instead of having to always compile from source - using node pre gyp just like sqlite3.

Reply:
okay, if they updated the installations steps, then we can left it in deps.

Reply:
I'm happy to move forward in whatever direction you think is best, here.  It's fine either way by me. The one thing I'd want to make sure is that there's at least a hint of somewhere with the expected version of `oracledb` so we can easily pull it in with CI without hard-coding the version in the CI config.